Source/AMRInterpolator/SurfaceExtraction.impl.hpp

```diff
@@ -25,6 +25,13 @@ SurfaceExtraction<SurfaceGeometry>::SurfaceExtraction(
       m_du(m_geom.du(m_params.num_points_u)),
       m_dv(m_geom.dv(m_params.num_points_v)), m_done_extraction(false)
 {
+    if (!FilesystemTools::directory_exists(m_params.data_path))
+        FilesystemTools::mkdir_recursive(m_params.data_path);
+
+    if (m_params.write_extraction &&
+        !FilesystemTools::directory_exists(m_params.extraction_path))
+        FilesystemTools::mkdir_recursive(m_params.extraction_path);
```

Comment:
I am a little uncomfortable with this being run every time a `SurfaceExtraction` object is instantiated (even though I doubt `stat` is slow). I can see why you didn't put this in `SimulationParametersBase` but could you add a `static bool` called e.g. `s_output_dirs_exist` to this class that starts as false and then changes to true once this has been done once and then condition these checks on that.

Reply:
I understand your concern. Not sure your suggestion works because there's no guarantee one won't use `SurfaceExtraction` with different parameters for different types of extraction (say Weyl extraction in one folder and ADM Quantities in another). Even thought it probably won't happen, it's not very secure in that sense. What about simply using the `a_first_step` variable? If it is the `first_step`, then do it. Would this work?